### PR TITLE
Update Nuget name and version to WindowsAppSDK preview version

### PR DESCRIPTION
**Pull request by erik0686**
I noticed many of the samples here were using nuget packages with the name ProjectReunion, when we should be using WindowsAppSDK instead. 

this PR updates the nuget packages to use WindowsAppSDK version 1.0.0-preview1.
Luckily, were in a good time as well because WindowsAppSDK version 1.0.0-preview1 also removes the need of adding the Foundation, WinUI, DWriteCore, and InteractiveExperiences nuget packages. 

To update, I used the visual studio UI and uninstalled the unneeded packages and installed the latest WindowsAppSDK package. 

Due to a renaming decision of ApplicationModel, I had to do a slight rename in some of the includes and namespaces, but after that the samples built correctly for all platforms and all configurations (Debug, Release, x64, x86 and ARM)

To review these changes, I built all the modified samples in all their configurations (Release, Debug) and platforms (ARM64, x64 and x86). I ran the samples as well and they all run correctly. 
Besides this, there might be some prerequisites you need to make sure you have done. You can find them here:
https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/deploy-unpackaged-apps

### CODE REVIEW

**Comment by jefgen** [comment]
Thanks for updating the Samples @erik0686! :)
(nit: Probably want to update the PR description on GitHub).

File: Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj [hunk 1]
```diff
@@ -1,9 +1,7 @@
...
-  <Import Project="packages\Microsoft.ProjectReunion.DWrite.0.8.0-preview\build\Microsoft.ProjectReunion.DWrite.props" Condition="Exists('packages\Microsoft.ProjectReunion.DWrite.0.8.0-preview\build\Microsoft.ProjectReunion.DWrite.props')" />
-  <Import Project="packages\Microsoft.ProjectReunion.0.8.0-preview\build\native\Microsoft.ProjectReunion.props" Condition="Exists('packages\Microsoft.ProjectReunion.0.8.0-preview\build\native\Microsoft.ProjectReunion.props')" />
+  <Import Project="packages\Microsoft.WindowsAppSDK.1.0.0-preview1\build\native\Microsoft.WindowsAppSDK.props" Condition="Exists('packages\Microsoft.WindowsAppSDK.1.0.0-preview1\build\native\Microsoft.WindowsAppSDK.props')" />
+  <Import Project="packages\Microsoft.Windows.SDK.BuildTools.10.0.20348.19\build\Microsoft.Windows.SDK.BuildTools.props" Condition="Exists('packages\Microsoft.Windows.SDK.BuildTools.10.0.20348.19\build\Microsoft.Windows.SDK.BuildTools.props')" />
```
Comment (jefgen) [inline Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj:4]: What is this new "Microsoft.Windows.SDK.BuildTools" package?

File: Samples/ResourceManagement/cpp-console-unpackaged/packages.config [hunk 2]
```diff
@@ -1,10 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.ProjectReunion" version="0.8.0-preview" targetFramework="native" />
-  <package id="Microsoft.ProjectReunion.DWrite" version="0.8.0-preview" targetFramework="native" />
-  <package id="Microsoft.ProjectReunion.Foundation" version="0.8.0-preview" targetFramework="native" />
-  <package id="Microsoft.ProjectReunion.InteractiveExperiences" version="0.8.0-preview" targetFramework="native" />
-  <package id="Microsoft.ProjectReunion.WinUI" version="0.8.0-preview" targetFramework="native" />
   <package id="Microsoft.Windows.CppWinRT" version="2.0.210714.1" targetFramework="native" />
   <package id="Microsoft.Windows.ImplementationLibrary" version="1.0.210204.1" targetFramework="native" />
+  <package id="Microsoft.Windows.SDK.BuildTools" version="10.0.20348.19" targetFramework="native" />
```
Comment (jefgen) [inline Samples/ResourceManagement/cpp-console-unpackaged/packages.config:5]: Same question about the "Microsoft.Windows.SDK.BuildTools" package.

File: Samples/TextRendering/cpp-win32/DWriteCoreGallery.sln [hunk 3]
```diff
@@ -21,9 +21,9 @@ Global
 		Release|x86 = Release|x86
 	EndGlobalSection
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
-		{D6EC0907-F515-4F61-93EC-B89E19FA58B4}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{D6EC0907-F515-4F61-93EC-B89E19FA58B4}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{D6EC0907-F515-4F61-93EC-B89E19FA58B4}.Debug|Any CPU.Deploy.0 = Debug|Any CPU
+		{D6EC0907-F515-4F61-93EC-B89E19FA58B4}.Debug|Any CPU.ActiveCfg = Debug|ARM64
```
Comment (jefgen) [inline Samples/TextRendering/cpp-win32/DWriteCoreGallery.sln:24]: This seems a bit odd... why is the "Any CPU" platform setting/building "ARM64"?

File: Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj (hunk 1 above)
Comment (erik0686) [reply Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj:4]: When adding WindowsAppSDK nuget package, it added it automatically. I assumed it was some kind of dependency, but I checked and the solution works correctly without it. I'll remove it :)

Comment (erik0686) [reply Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj:4]: It is a dependency from Microsoft.WindowsAppSDK. If I try to remove it, I am shown the following error:
```Unable to uninstall 'Microsoft.Windows.SDK.BuildTools.10.0.20348.19' because 'Microsoft.WindowsAppSDK.1.0.0-preview1' depends on it.```

File: Samples/TextRendering/cpp-win32/DWriteCoreGallery.sln (hunk 3 above)
Comment (erik0686) [reply Samples/TextRendering/cpp-win32/DWriteCoreGallery.sln:24]: Weird.... I must have misclicked somewhere. I'll fix

File: Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj (hunk 1 above)
Comment (rohanp-msft) [reply Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj:4]: Microsoft.Windows.SDK.BuildTools was added to WAS recently. It removes the dependency on the Windows SDK installed on the machine for build tools like makepri.exe, makeappx.exe, etc.

Comment (jefgen) [reply Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj:4]: Thanks @rohanp-msft! :)

File: Samples/ResourceManagement/cs-winui/winui_desktop_packaged_app/winui_desktop_packaged_app.csproj [hunk 4]
```diff
@@ -21,9 +21,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.ProjectReunion" Version="0.8.1" />
-    <PackageReference Include="Microsoft.ProjectReunion.Foundation" Version="0.8.1" />
-    <PackageReference Include="Microsoft.ProjectReunion.WinUI" Version="0.8.1" />
+    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.0.0-preview1" />
```
Comment (rohanp-msft) [inline Samples/ResourceManagement/cs-winui/winui_desktop_packaged_app/winui_desktop_packaged_app.csproj:24]: I would have expected to also see Microsoft.Windows.SDK.BuildTools listed here... Same for the other CSPROJ and WAPPROJ files. You can see that the package shows up in the VCXPROJ files.

Can you check in NuGet Package Manager whether these projects have Microsoft.Windows.SDK.BuildTools installed?

File: Samples/ResourceManagement/cs-winforms-unpackaged/winforms_unpackaged_app.csproj [hunk 5]
```diff
@@ -10,7 +10,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.ProjectReunion" Version="0.8.0-preview" />
+    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.0.0-preview1" />
```
Comment (rohanp-msft) [inline Samples/ResourceManagement/cs-winforms-unpackaged/winforms_unpackaged_app.csproj:13]: In case you didn't know: verifying unpackaged apps requires some additional steps. Please see https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/deploy-unpackaged-apps#prerequisites.

File: Samples/TextRendering/cpp-win32/DWriteCoreGallery/DWriteCoreGallery.vcxproj [hunk 6]
```diff
@@ -1,8 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="..\packages\Microsoft.ProjectReunion.0.8.1\build\native\Microsoft.ProjectReunion.props" Condition="Exists('..\packages\Microsoft.ProjectReunion.0.8.1\build\native\Microsoft.ProjectReunion.props')" />
-  <Import Project="..\packages\Microsoft.ProjectReunion.Foundation.0.8.1\build\native\Microsoft.ProjectReunion.Foundation.props" Condition="Exists('..\packages\Microsoft.ProjectReunion.Foundation.0.8.1\build\native\Microsoft.ProjectReunion.Foundation.props')" />
-  <Import Project="..\packages\Microsoft.ProjectReunion.DWrite.0.8.1\build\Microsoft.ProjectReunion.DWrite.props" Condition="Exists('..\packages\Microsoft.ProjectReunion.DWrite.0.8.1\build\Microsoft.ProjectReunion.DWrite.props')" />
```
Comment (rohanp-msft) [inline Samples/TextRendering/cpp-win32/DWriteCoreGallery/DWriteCoreGallery.vcxproj:5]: Since you're taking on DWriteCore as well in this change, please add DWriteCore folks to the review: Duncan MacMichael and Niklas Borson.

**Comment by rohanp-msft** [comment]
Would you please add a How Verified section to your PR description? It would help the reviewers to know which cases have been verified, if there are any gaps, etc.

File: Samples/ResourceManagement/cs-winui/winui_desktop_packaged_app/winui_desktop_packaged_app.csproj (hunk 4 above)
Comment (erik0686) [reply Samples/ResourceManagement/cs-winui/winui_desktop_packaged_app/winui_desktop_packaged_app.csproj:24]: Hmmm interesting... 
If I uninstall and install again Microsoft.WindowsAppSDK, I get this:
```
Successfully installed 'Microsoft.Windows.SDK.BuildTools 10.0.20348.19' to winui_desktop_packaged_app
Successfully installed 'Microsoft.WindowsAppSDK 1.0.0-preview1' to winui_desktop_packaged_app
```
However, I don't see the SDK BuildTools installed after that, and they also don't appear in the csproj files and others. That said, the solution still builds correctly in all platforms and configurations. 

Is it possible that VS determines that it doesn't need the BuildTools? Can that happen?

Comment (rohanp-msft) [reply Samples/ResourceManagement/cs-winui/winui_desktop_packaged_app/winui_desktop_packaged_app.csproj:24]: > Is it possible that VS determines that it doesn't need the BuildTools? Can that happen?
AFAIK, no...

Here's my suggestion. Build the app in VS Developer Command Prompt with /bl. Then, search for makepri.exe in https://msbuildlog.com/. See which makepri.exe was used.

**Comment by huichen123** [comment]
>     // Take a dependency on Project Reunion v0.8 preview.

still 0.8

---
Refers to: Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.cpp:19 in c45ca8c. [](commit_id = c45ca8c5ede0c506837cba4492bb114dbe411597, deletion_comment = False)

File: Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj (hunk 1 above)
Comment (huichen123) [inline Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj:4]: >Microsoft.Windows.SDK.BuildTools.props" [](http://example.com/codeflow?start=81&length=39)

do we still need this?

File: Samples/ResourceManagement/cs-winui/winui_desktop_packaged_app/winui_desktop_packaged_app.csproj (hunk 4 above)
Comment (huichen123) [reply Samples/ResourceManagement/cs-winui/winui_desktop_packaged_app/winui_desktop_packaged_app.csproj:24]: i think you don't need the explicit reference to buildtools with PackageReference.

Comment (erik0686) [reply Samples/ResourceManagement/cs-winui/winui_desktop_packaged_app/winui_desktop_packaged_app.csproj:24]: that makes sense, I do see the makepri.exe of the Buildtools being used, but I just can't seem to see it in packageReference. Do you know why's this, @huichen123 ?

File: Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.cpp [hunk 7]
```diff
@@ -7,16 +7,16 @@
...
 using namespace winrt;
-using namespace winrt::Microsoft::ApplicationModel::Resources;
+using namespace winrt::Microsoft::Windows::ApplicationModel::Resources;
 
 HRESULT LoadProjectReunion()
 {
-    // Take a dependency on Project Reunion v0.8 preview.
+    // Take a dependency on WindowsAppSDK 1.0 preview.
     const UINT32 majorMinorVersion{ 0x00000008 };
     PCWSTR versionTag{ L"preview" };
```
Comment (huichen123) [inline Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.cpp:21]: need to change code as well.

File: Samples/ResourceManagement/cs-winui/ClassLibrary/ClassLibrary.csproj [hunk 8]
```diff
@@ -10,7 +10,8 @@
     <None Remove="ClassLibraryResources.resw" />
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="Microsoft.ProjectReunion" Version="0.8.1" />
+    <PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.20348.19" />
```
Comment (huichen123) [inline Samples/ResourceManagement/cs-winui/ClassLibrary/ClassLibrary.csproj:13]: ><PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.20348.19" /> [](http://example.com/codeflow?start=4&length=87)

is this necessary?

**Review by niklasb-ms — APPROVED** [review]

File: Samples/ResourceManagement/cs-winui/ClassLibrary/ClassLibrary.csproj (hunk 8 above)
Comment (erik0686) [reply Samples/ResourceManagement/cs-winui/ClassLibrary/ClassLibrary.csproj:13]: Well, for this project, when updating the WindowsAppSDK nuget package to version 1.0 preview, it automatically adds the BuildTools. You mentioned in another comment that the PackageReference might not be necessary, and after trying it out without the BuildTools line, it does work. However, since VS insists in adding it, do you think I should still remove it?

Comment (huichen123) [reply Samples/ResourceManagement/cs-winui/ClassLibrary/ClassLibrary.csproj:13]: If VS adds it, leave it there. I didn't know you manually removed BuildTools reference in your previous iterations.

Comment (erik0686) [reply Samples/ResourceManagement/cs-winui/ClassLibrary/ClassLibrary.csproj]: I'll add them back

File: Samples/Windowing/cpp-winui/SampleApp/MainWindow.xaml.cpp [hunk 9]
```diff
@@ -87,7 +87,7 @@ namespace winrt::SampleApp::implementation
 
         // Get the WindowId for our window
         winrt::WindowId windowId;
-        winrt::GetWindowIdFromWindowHandle(hWnd, &windowId);
```
Comment (rohanp-msft) [inline Samples/Windowing/cpp-winui/SampleApp/MainWindow.xaml.cpp:90]: Curious, why this change?

File: Samples/ResourceManagement/cs-winforms-unpackaged/Program.cs [hunk 10]
```diff
@@ -23,7 +23,7 @@ static void Main()
             // Initialize dynamic dependencies so we can consume the Project Reunion APIs in the Project
             // Reunion framework package from this unpackaged app.
             // Take a dependency on Project Reunion v0.8 preview.
```
Comment (rohanp-msft) [inline Samples/ResourceManagement/cs-winforms-unpackaged/Program.cs:25]: >Project Reunion v0.8 preview [](http://example.com/codeflow?start=36&length=28)

Needs update.

File: Samples/ResourceManagement/cs-winforms-unpackaged/Program.cs [hunk 11]
```diff
@@ -23,7 +23,7 @@ static void Main()
             // Initialize dynamic dependencies so we can consume the Project Reunion APIs in the Project
```
Comment (rohanp-msft) [inline Samples/ResourceManagement/cs-winforms-unpackaged/Program.cs:23]: >Project Reunion [](http://example.com/codeflow?start=69&length=15)

Would be good to change this too, like you did in the other unpackaged app.

File: Samples/Windowing/cpp-winui/SampleApp/MainWindow.xaml.cpp (hunk 9 above)
Comment (erik0686) [reply Samples/Windowing/cpp-winui/SampleApp/MainWindow.xaml.cpp:90]: Honestly, I'm not sure if this is something that changed between versions of WindowsAppSDK, but after updating to WAS 1.0.0-preview, this sample would not build correctly, since it was saying "GetWindowIdFromWindowHandle() didn't take two parameters." I saw the definition of that API and it indeed only took 1 parameter, and returned a WindowID. 

I figured since it wasn't a huge change, I might do it so it would build and run correctly :)

File: Samples/ResourceManagement/cs-winforms-unpackaged/Program.cs [hunk 12]
```diff
@@ -23,7 +23,7 @@ static void Main()
             // Initialize dynamic dependencies so we can consume the Project Reunion APIs in the Project
             // Reunion framework package from this unpackaged app.
```
Comment (rohanp-msft) [inline Samples/ResourceManagement/cs-winforms-unpackaged/Program.cs:24]: Same for the second instance.

File: Samples/Windowing/cpp-winui/SampleApp/MainWindow.xaml.cpp (hunk 9 above)
Comment (rohanp-msft) [reply Samples/Windowing/cpp-winui/SampleApp/MainWindow.xaml.cpp:90]: Got it! Yeah, good to do!

Since you're also modifying the Windowing samples, adding Samantha Song and Eve Mwangi would be good.

**Review by samanthamsong — APPROVED** [review]

File: Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.cpp [hunk 13]
```diff
@@ -58,11 +59,11 @@ int wmain(int argc, wchar_t* argv[])
...
 
-    // Initialize dynamic dependencies so we can consume the Project Reunion APIs in the Project Reunion framework package from this unpackaged app. 
-    HRESULT loadProjectReunionHr = LoadProjectReunion();
-    if (FAILED(loadProjectReunionHr))
+    // Initialize dynamic dependencies so we can consume the WindowsAppSDK APIs in the WindowsAppSDK framework package from this unpackaged app. 
+    HRESULT loadWindowsAppSDKHr = LoadWindowsAppSDK();
+    if (FAILED(loadWindowsAppSDKHr))
     {
-        std::wcout << "Could not load Project Reunion!" << std::endl;
+        std::wcout << "Could not load WindowsAppSDK!" << std::endl;
```
Comment (rohanp-msft) [inline Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.cpp:66]: >WindowsAppSDK [](http://example.com/codeflow?start=38&length=13)

A comment for the change overall: Windows App SDK in text (whether in error messages or comments or whatever else) shouldn't be in CamelCase. The name of the product is "Windows App SDK". The CamelCase version can be for code of course.

**Review by rohanp-msft — APPROVED** [review]
Just one new comment for commit 13. Otherwise change looks good. Thanks again.
